### ISSUE TITLE
After setting setLocalPressure wrong altitude is returned

### ISSUE THREAD

**Issue by ryansommers** [dpyeates/react-native-barometer#2]
Tried to submit a PR but realized I don't know what I'm doing.

The issue: when running setLocalPressure(968.76) with the Barometer reporting hPa of 978.7792 I get a final altitude of -93. I'd expect it to be more in the ~390 range.

**Comment by ryansommers** [comment]
Think this is user error. It's returning -93 as the difference between reported value and local value. If I add 93 to the ASL altitude (291), it gets close to what I expect. Is that the intention for how to use this?

**Comment by dpyeates** [comment]
Hi Ryan, let me look into this today for you.

**Comment by dpyeates** [comment]
Hi Ryan.

I've run your figures through an online calculator and the figure of around -93 is correct.

Try this link and enter 968.76 as sea-level pressure (Po) 978.7792 as your atmospheric pressure then hit execute. You will get -86.68m. This figure is different as this website uses the 'hypsometric formula' (which allows for variation in temperature) vs. the 'barometric formula' which is what i use in my library.
 
https://keisan.casio.com/exec/system/1224585971

Additional info:
https://physics.stackexchange.com/questions/333475/how-to-calculate-altitude-from-current-temperature-and-pressure

What you get returned by the library is:

"pressure": the observed/sensed local air pressure
"altitudeASL": your current altitude above sea level based upon standard atmosphere in m
"altitude": your current altitude based upon the custom local pressure that you have applied in m
"relativeAltitude": the diff in me between when you started and current time +/- in m
"verticalSpeed": your current estimated vertical speed in m/s +/-

**Comment by ryansommers** [comment]
Hey Darren, thanks for taking the time to reply on this!

I might just be thinking about it wrong, but what I was trying to do is this:

1) The barometer was reporting an altitude of ~290m based on ASL
2) After waiting for GPS altitude accuracy to be < 3m, it was giving me an altitude of ~390, which is correct for where I live
3) I used the calculator at https://keisan.casio.com/exec/system/1224579725 to figure out an hPa of 968.76
4) I used setLocalPressure(968.76) and the "altitude" returned was -93, which I realize now is the difference between the local pressure (968.76) and the pressure the Barometer was reporting ( 978.78)
5) If I add 93m to the ~290m reported, I get pretty close to my expected altitude. But what happens if the reported pressure was ~968, making the difference 0, and the Barometer still thinks my altitude from ASL is ~290m - thus being 100m off? Maybe that technically won't happen - like I said, I don't really understand all this barometric calculation stuff, I just want an accurate altitude in the end ;)
6) What I guess I'd hoped would happen is after using setLocalPressure(), it'd give me a more accurate altitude from sea level - basically close to what's being reported by GPS.
